modules/afamqp/afamqp.c

```diff
@@ -463,8 +488,28 @@ afamqp_dd_connect(AMQPDestDriver *self, gboolean reconnect)
       goto exception_amqp_dd_connect_failed_init;
     }
 
-  ret = amqp_login(self->conn, self->vhost, self->max_channel, self->frame_size, 0,
-                   AMQP_SASL_METHOD_PLAIN, self->user, self->password);
+  switch (self->auth_method)
+    {
+    case AMQP_SASL_METHOD_PLAIN:
+      ret = amqp_login(self->conn, self->vhost, self->max_channel, self->frame_size, 0,
+                       self->auth_method, self->user, self->password);
+      break;
+
+    case AMQP_SASL_METHOD_EXTERNAL:
+      // The identity is generally not needed for external authentication, but must be set to
+      // something non-empty otherwise the API call will fail, so just default it to something
+      // meaningless
+      identity = (self->user && self->user[0] != '\0') ? self->user : ".";
+      ret = amqp_login(self->conn, self->vhost, self->max_channel, self->frame_size, 0,
+                       self->auth_method, identity);
+      break;
+
+    default:
```

Comment:
Since we validate when the grammar is parsed, this can easily become an assertion. No need to handle errors as they will essentially be dead code. Just use g_assert_not_reached ()

Reply:
Travis compiles with `-Wswitch-default`, so it is kinda forced by us.